File: git/labler/labler.py
import json

import requests

groupUrl = "https://gitlab.com/api/v4/groups/{group}/projects?private_token={token}"
labelsUrl = "https://gitlab.com/api/v4/projects/{project_id}/labels?private_token={token}"


def main():
    print("Enter GitLab Private Token:")
    key = input()
    key = key.strip()

    print("Enter GitLab Group Name:")
    group = input()

    if (not group):
        print("group should not be empty")
        return

    url = groupUrl.replace("{group}", group)
    url = url.replace("{token}", key)

    answer = requests.get(url)

    if (answer.status_code != 200):
        print(answer.json()['message'] + " for group: " + group)
        return

    projects = answer.json()

    ids = []
    for count, project in enumerate(projects):
        name = "%2d %20s: %8d" % (count + 1, project['name'], project['id'])
        ids.append(project['id'])
        print(name)

    if (len(ids) < 1):
        print("No Repository found for group: " + group)

    print("Choose Repository:")
    choosedId = int(input()) - 1
    repositoryId = ids[choosedId]
    url = labelsUrl.replace('{project_id}', str(repositoryId))
    url = url.replace("{token}", key)
    answer = requests.get(url)
    if (answer.status_code == 200):
        with open('labels.json') as file:
            labels = json.load(file)
        for label in labels:
            answer = requests.post(url, {"name": label[0], "color": label[1]}).json()
            if ('name' in answer or 'color' in answer):
                print("Label %10s with color %10s created" % (answer['name'], answer['color']))
            elif ('message' in answer):
                print(answer['message'] + ": " + label[0])
            else:
                print(answer)


if __name__ == "__main__":
    main()

File: git/labler/labler.php
<?php


$url = 'https://gitlab.com/api/v4/projects/{project_id}/labels';
$url = str_replace('{project_id}', 8697203, $url);
$ch = curl_init();
curl_setopt($ch,CURLOPT_URL, $url);
curl_setopt($ch,CURLOPT_RETURNTRANSFER, true); 
curl_setopt($ch, CURLOPT_HEADER, false);
curl_setopt($ch, CURLOPT_HTTPHEADER, ['PRIVATE-TOKEN: token']);



$labels = json_decode(file_get_contents(__DIR__.'/labels.json',true));

foreach ($labels as $label) {
	var_dump($url);
	$fields = ['name' => $label[0],'color' => $label[1]];
$fields_string = http_build_query($fields);
curl_setopt($ch,CURLOPT_POSTFIELDS, $fields_string);
curl_setopt($ch,CURLOPT_POST, count($fields));
$result = curl_exec($ch);
var_dump($result);

}

